apps/web/src/components/Basenames/UsernameProfileSectionFrames/index.tsx

```diff
@@ -0,0 +1,112 @@
+'use client';
+
+import { useAnalytics } from 'apps/web/contexts/Analytics';
+import { useUsernameProfile } from 'apps/web/src/components/Basenames/UsernameProfileContext';
+import {
+  FramesProvider,
+  useFrameContext,
+} from 'apps/web/src/components/Basenames/UsernameProfileSectionFrames/Context';
+import FarcasterAccountModal from 'apps/web/src/components/Basenames/UsernameProfileSectionFrames/FarcasterAccountModal';
+import FrameListItem from 'apps/web/src/components/Basenames/UsernameProfileSectionFrames/FrameListItem';
+import UsernameProfileSectionTitle from 'apps/web/src/components/Basenames/UsernameProfileSectionTitle';
+import { Button, ButtonSizes } from 'apps/web/src/components/Button/Button';
+import ImageAdaptive from 'apps/web/src/components/ImageAdaptive';
+import { ActionType } from 'libs/base-ui/utils/logEvent';
+import { StaticImageData } from 'next/image';
+import Link from 'next/link';
+import { useCallback } from 'react';
+import cornerGarnish from './corner-garnish.svg';
+import frameIcon from './frame-icon.svg';
+
+function SectionContent() {
+  const { profileUsername, currentWalletIsProfileOwner } = useUsernameProfile();
+  const {
+    frameInteractionError,
+    setFrameInteractionError,
+    frameUrls,
+    existingTextRecordsIsLoading,
+  } = useFrameContext();
+  const handleErrorClick = useCallback(
+    () => setFrameInteractionError(''),
+    [setFrameInteractionError],
+  );
+  const { logEventWithContext } = useAnalytics();
+  const handleAddFrameLinkClick = useCallback(() => {
+    logEventWithContext('basename_profile_frame_try_now_clicked', ActionType.click);
+  }, [logEventWithContext]);
+
+  if (currentWalletIsProfileOwner && frameUrls.length === 0 && !existingTextRecordsIsLoading) {
+    return (
```

Comment:
Would personally extract this to a component since it seems quite static and is just a huge blob of JSX here.